src/main/java/com/airlines/dao/impl/AirplaneDaoImpl.java

```diff
@@ -78,60 +81,95 @@ public AirplaneDaoImpl(DataSource dataSource) {
                         .build());
             }
         } catch (SQLException e) {
-            throw new DaoOperationException("Couldn't save product ", e);
+            throw new DaoOperationException("Couldn't find an airplane ", e);
         }
         return airplaneArray;
     }
 
     @Override
-    public Airplane findOne(String codeName) {
-        Airplane airplane = null;
+    public Optional<Airplane> findByCodeName(String codeName) {
         try (Connection connection = dataSource.getConnection();
-             PreparedStatement selectByIdStatement = connection.prepareStatement(SELECT_BY_CODE_NAME_SQL);) {
-            selectByIdStatement.setString(1, codeName);
-            ResultSet resultSet = selectByIdStatement.executeQuery();
+             PreparedStatement selectByCodeNameStatement = connection.prepareStatement(SELECT_BY_CODE_NAME_QUERY);) {
+            selectByCodeNameStatement.setString(1, codeName);
+            ResultSet resultSet = selectByCodeNameStatement.executeQuery();
             if (resultSet.next()) {
-                airplane = Airplane.builder()
+                return Optional.of(Airplane.builder()
                         .withId(resultSet.getLong("id"))
                         .withCodeName(resultSet.getString("name"))
                         .withModel(resultSet.getString("model"))
                         .withManufactureDate(resultSet.getDate("manufactureDate").toLocalDate())
                         .withCapacity(resultSet.getInt("capacity"))
                         .withFlightRange(resultSet.getInt("flightRange"))
                         .withCrew((Crew) resultSet.getObject("crew"))
-                        .build();
+                        .build());
+            } else {
+                return Optional.empty();
             }
-        } catch (SQLException throwables) {
-            throwables.printStackTrace();
+        } catch (SQLException e) {
+            throw new DaoOperationException("Couldn't find an airplane ", e);
         }
-        return airplane;
     }
 
     @Override
-    public void remove(Airplane airplane) {
-        if (airplane.getId() == null) {
-            throw new DaoOperationException("Cannot find a product without ID");
-        }
-        if (airplane.getId() < 0) {
-            throw new DaoOperationException(String.format("Product with id = %d does not exist", airplane.getId()));
+    public void remove(Long id) {
+        if (id == null) {
+            throw new DaoOperationException("Cannot remove an airplane without id");
         }
         try (Connection connection = dataSource.getConnection();
-             PreparedStatement statement = connection.prepareStatement(DELETE_AIRPLANE_SQL);) {
-            statement.setLong(1, airplane.getId());
+             PreparedStatement statement = connection.prepareStatement(DELETE_AIRPLANE_QUERY);) {
+            statement.setLong(1, id);
             statement.executeUpdate();
```

Comment:
DELETE query structurally and technically almost the same as SELECT query.
It receives no data.
It consist usually 1 attribute (for simple select query for example)

in findByCodeName you used 
`ResultSet resultSet = selectByCodeNameStatement.executeQuery();`
in remove you are using:
`statement.executeUpdate();`
I recommend using executeQuery in this case.

You can make time, for example 45 min, to investigate next topic:
- difference between Statement and PreparedStatement
- difference between execute(), executeUpdate(), executeQuery() methods